league/battle.py

```diff
@@ -23,29 +23,85 @@ def draw(self, screen: pygame.Surface) -> None:
+        self.map_pos_x = -100
+        self.map_pos_y = 100
+
+        self.map = pygame.image.load('league\\images\\map.png')
+
+        # taken from png properties
+        self.MAP_WIDTH = 2403
+        self.MAP_HEIGHT = 5120
+        self.MAP_SIZE = (self.MAP_WIDTH, self.MAP_HEIGHT)
+
+        # Grid variables
+        pygame.font.init()
+
+        self.font = pygame.font.SysFont('Calibri', 50)
+
+        self.ROWS = 20
+        self.COLUMNS = 5
+
+        self.cell_width = self.MAP_WIDTH // self.COLUMNS
+        self.cell_height = self.MAP_HEIGHT // self.ROWS
+
 
     def event_loop(self, events: List[pygame.event.Event]) -> None:
         for event in events:
             if event.type == KEYDOWN:
                 print(event)
 
-    def update(self) -> None:
+
+    def update(self, screen) -> None:
         keys_pressed = pygame.key.get_pressed()
         if keys_pressed[K_d]:
-            self.circle_x += 5
+            color = screen.get_at((self.pos_x + self.PLAYER_RADIUS + 1, self.pos_y))
+            if self.pos_x < self.MAP_WIDTH and color != (0,0,0,255): # should check for blue and red team?
+                self.map_pos_x -= self.speed
         
         if keys_pressed[K_a]:
-            self.circle_x -= 5
+            color = screen.get_at((self.pos_x - self.PLAYER_RADIUS - 1, self.pos_y))
+            if self.pos_x > 0 and color != (0,0,0,255):
+                self.map_pos_x += self.speed
 
         if keys_pressed[K_s]:
-            self.circle_y += 5
-        
+            color = screen.get_at((self.pos_x, self.pos_y + self.PLAYER_RADIUS + 1))
+            if self.pos_y < self.MAP_HEIGHT and color != (0,0,0,255):
+                self.map_pos_y -= self.speed
+
         if keys_pressed[K_w]:
-            self.circle_y -= 5
+            color = screen.get_at((self.pos_x, self.pos_y - self.PLAYER_RADIUS - 1))
+            if self.pos_y > 0 and color != (0,0,0,255):
+                self.map_pos_y += self.speed
         
     
     def draw(self, screen: pygame.Surface) -> None:
-        screen.fill((255, 255, 255))
-        pygame.draw.circle(screen, (0, 0, 128), (self.circle_x, self.circle_y), 20)
\ No newline at end of file
+        screen.fill((0,0,0))
+
+        # draw grid
+        for x in range(0, self.MAP_WIDTH, self.cell_width):
+            pygame.draw.line(self.map, (128,128,128), (x,0), (x, self.MAP_HEIGHT), 1)
+
+        for y in range(0, self.MAP_HEIGHT, self.cell_height):
+            pygame.draw.line(self.map, (128,128,128), (0,y), (self.MAP_WIDTH, y), 1)
+
+        # add labels to each cell
+        row_letters = list("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+        for i, x in enumerate(range(0, self.MAP_WIDTH, self.cell_width)):
+            for j, y in enumerate(range(0, self.MAP_HEIGHT, self.cell_height)):
+                text = row_letters[j] + str(i)
+
+                grid_text = self.font.render(text, True, (0, 0, 255))
+                self.map.blit(grid_text, (x, y))
```

Comment:
Create a private method to draw this. `_draw_grid`. This way we can easily comment it out here in the draw method.